app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsCorrectQtyCondition.php

```diff
@@ -0,0 +1,158 @@
+     */
+    private $getStockItemConfiguration;
+
+    /**
+     * @var GetReservationsQuantityInterface
+     */
+    private $getReservationsQuantity;
+
+    /**
+     * @var GetStockItemDataInterface
+     */
+    private $getStockItemData;
+
+    /**
+     * @var StockConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var MathDivision
+     */
+    protected $mathDivision;
+
+    /**
+     * @var ProductSalabilityErrorFactory
+     */
+    private $productSalabilityErrorFactory;
+
+    /**
+     * @var IsProductSalableResultFactory
+     */
+    private $isProductSalableResultFactory;
+
+    public function __construct(
+        GetStockItemConfigurationInterface $getStockItemConfiguration,
+        StockConfigurationInterface $configuration,
+        GetReservationsQuantityInterface $getReservationsQuantity,
+        GetStockItemDataInterface $getStockItemData,
+        MathDivision $mathDivision,
+        ProductSalabilityErrorFactory $productSalabilityErrorFactory,
+        IsProductSalableResultFactory $isProductSalableResultFactory
+    ) {
+        $this->getStockItemConfiguration = $getStockItemConfiguration;
+        $this->configuration = $configuration;
+        $this->getStockItemData = $getStockItemData;
+        $this->getReservationsQuantity = $getReservationsQuantity;
+        $this->mathDivision = $mathDivision;
+        $this->productSalabilityErrorFactory = $productSalabilityErrorFactory;
+        $this->isProductSalableResultFactory = $isProductSalableResultFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(string $sku, int $stockId, float $requestedQty): IsProductSalableResultInterface
+    {
+        $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);
+        if (null === $stockItemConfiguration) {
+            $errors = [
+                $this->productSalabilityErrorFactory->create([
+                    'code' => 'is_correct_qty-no_config',
+                    'message' => __('Missing stock item configuration')
+                ])
+            ];
+            return $this->isProductSalableResultFactory->create(['errors' => $errors]);
+        }
+
+        // Out-of-Stock Threshold
+        // TODO verify whether we should use < or <=
+        $globalMinQty = $this->configuration->getMinQty();
+        if (($stockItemConfiguration->isUseConfigMinQty() == 1 && $requestedQty < $globalMinQty)
+            || ($stockItemConfiguration->isUseConfigMinQty() == 0
+                && $requestedQty < $stockItemConfiguration->getMinQty()
+            )) {
+            $errors = [
+                $this->productSalabilityErrorFactory->create([
+                    'code' => 'is_correct_qty-out_of_stock_threshold',
+                    'message' => __('The requested qty is not available')
+                ])
+            ];
+            return $this->isProductSalableResultFactory->create(['errors' => $errors]);
+        }
```

Comment:
Hey @aleron75 ,

I was looking at your PR and looks like you have an error in this check
```php
        // Out-of-Stock Threshold
        // TODO verify whether we should use < or <=
        $globalMinQty = $this->configuration->getMinQty();
        if (($stockItemConfiguration->isUseConfigMinQty() == 1 && $requestedQty < $globalMinQty)
            || ($stockItemConfiguration->isUseConfigMinQty() == 0
                && $requestedQty < $stockItemConfiguration->getMinQty()
            )) {
            $errors = [
                $this->productSalabilityErrorFactory->create([
                    'code' => 'is_correct_qty-out_of_stock_threshold',
                    'message' => __('The requested qty is not available')
                ])
            ];
            return $this->isProductSalableResultFactory->create(['errors' => $errors]);
        }
```

you make it the same as `getMinSaleQty`, but these configuration options are different

`getMinSaleQty` - represents minimal amount in the shopping cart, while `MinQty` is responsible for minimal Qty available in Stock

for example, in Legacy Magento we do this check :
`if ($stockItem->getQty() - $stockItem->getMinQty() - $qty < 0) { .. }`
something similar should be here as well

Reply:
yes, you are right, I misunderstood the logic, will fix it